### PR TITLE
EVG-5407 Update patch task CreateTime to be base commit date

### PR DESCRIPTION
**Pull request by syev**

### CODE REVIEW

File: model/lifecycle.go [hunk 1]
```diff
@@ -845,9 +851,26 @@ func TryMarkPatchBuildFinished(b *build.Build, finishTime time.Time, updates *St
...
 
+func getTaskCreateTime(project *Project, v *version.Version) (*time.Time, error) {
+	if evergreen.IsPatchRequester(v.Requester) {
+		baseVersion, err := version.FindOne(version.BaseVersionFromPatch(project.Identifier, v.Revision))
+		if err != nil {
+			grip.Error(errors.Wrap(err, "Error finding base version for patch version"))
+			return nil, err
+		} else if baseVersion == nil {
+			err = errors.Errorf("Could not find base version for patch version %s", v.Id)
+			grip.Warning(err)
```
Comment (tychoish) [inline model/lifecycle.go]: probably not worth logging and returning errors.

File: model/lifecycle.go [hunk 2]
```diff
@@ -681,7 +681,10 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 			}
 			execTaskIds = append(execTaskIds, execTaskId)
 		}
-		t := createDisplayTask(id, dt.Name, execTaskIds, buildVariant, b, v, project)
+		t, err := createDisplayTask(id, dt.Name, execTaskIds, buildVariant, b, v, project)
+		if err != nil {
+			return tasks, err
```
Comment (tychoish) [inline model/lifecycle.go]: probably would be a good idea to wrap/annotate this and the reset of the errors added in this file

File: model/lifecycle.go [hunk 3]
```diff
@@ -845,9 +851,26 @@ func TryMarkPatchBuildFinished(b *build.Build, finishTime time.Time, updates *St
 	return nil
 }
 
+func getTaskCreateTime(project *Project, v *version.Version) (*time.Time, error) {
+	if evergreen.IsPatchRequester(v.Requester) {
+		baseVersion, err := version.FindOne(version.BaseVersionFromPatch(project.Identifier, v.Revision))
+		if err != nil {
+			grip.Error(errors.Wrap(err, "Error finding base version for patch version"))
+			return nil, err
+		} else if baseVersion == nil {
```
Comment (tychoish) [inline model/lifecycle.go]: it's possible for these db functions to return nil errors and nil data, so I think it's better to not use an else.

File: model/lifecycle.go [hunk 4]
```diff
@@ -845,9 +851,26 @@ func TryMarkPatchBuildFinished(b *build.Build, finishTime time.Time, updates *St
 	return nil
 }
 
+func getTaskCreateTime(project *Project, v *version.Version) (*time.Time, error) {
```
Comment (tychoish) [inline model/lifecycle.go]: why return a pointer to time.Time? It seems more complicated than just return the concrete value.

File: model/lifecycle.go (hunk 1 above)
Comment (syev) [reply model/lifecycle.go]: I removed the logging for these errors.

File: model/lifecycle.go (hunk 3 above)
Comment (syev) [reply model/lifecycle.go]: I'm not sure I understand what you mean.
The check in the "else if" is handling the case you mention of nil error and nil data.
Also the if clause returns, so in this case the behavior is the same with or without the else clause.

File: model/lifecycle.go (hunk 4 above)
Comment (syev) [reply model/lifecycle.go]: Right, I used a pointer to return nil, I have to get used to returning empty structs. Done.

File: model/lifecycle.go (hunk 2 above)
Comment (syev) [reply model/lifecycle.go]: Good point, thanks. Done.

File: model/lifecycle.go (hunk 3 above)
Comment (john-m-liu) [reply model/lifecycle.go]: The else if doesn't run if err is nil

File: model/lifecycle.go [hunk 5]
```diff
@@ -845,9 +852,24 @@ func TryMarkPatchBuildFinished(b *build.Build, finishTime time.Time, updates *St
 	return nil
 }
 
+func getTaskCreateTime(project *Project, v *version.Version) (time.Time, error) {
```
Comment (john-m-liu) [inline model/lifecycle.go]: I would have this take in a project ID as a string since you don't use any other properties of it

**Pull request merged**